push-sender/src/main/java/org/jboss/aerogear/unifiedpush/message/jms/AbstractJMSMessageListener.java

```diff
@@ -35,16 +35,15 @@
  */
 public abstract class AbstractJMSMessageListener<T> implements MessageListener {
 
-    @Resource(mappedName = "java:/ConnectionFactory")
-    private ConnectionFactory connectionFactory;
-
     private static final Logger logger = LoggerFactory.getLogger(AbstractJMSMessageListener.class);
 
     public abstract void onMessage(T message);
 
     @SuppressWarnings("unchecked")
     @Override
+    @TransactionAttribute(TransactionAttributeType.REQUIRED)
     public void onMessage(Message jmsMessage) {
+        logger.debug("message get " + jmsMessage.toString());
```

Comment:
Change this to `logger.debug("message get {}", jmsMessage)` to avoid the string concatenation to happens when the debug messages are disabled.